src/app/clusters/mode-base-server/readme.md

```diff
@@ -0,0 +1,35 @@
+# Mode Base and its aliases
+
+ModeBase is a pseudo cluster. It has no cluster ID. It exists only to be derived
+into other clusters.
+
+# How to use a ModeBase derived cluster
+
+To use a ModeBase derived cluster, you need to
+
+-   create a class that inherits the `ModeBase::Instance` class. For examples
+    look at the `<alias name>-mode.x` files in the `src` and `include`
+    directorins in `examples/all-clusters-app/all-clusters-common`.
+-   For this class implement the `NumberOfModes`, `GetModeLabelByIndex`,
+    `GetModeValueByIndex` `GetModeTagsByIndex` and `HandleChangeToMode`
+    functions. Look at the documentation in `mode-base-server.h` for more
+    information about these functions. Cluster specific enums can be accessed
+    from the cluster namespace.
+-   Optionally implement the `AppInit` function.
+-   In your main file, instantiate your `ModeBase::Instance` inherited class.
+    See the constructor documentation in `mode-base-server.h`.
+-   Call the `.Init()` function of your instance.
```

Comment:
When? After `Server::Init()` ?

Reply:
What do you mean by `server::init()`? The instance here runs the server code so `<my instance>.init()` is initialising a ModeBase server.

Reply:
By Server::Init(), I mean the root of all server stuff (src/app/server/Server.cpp)

Reply:
Updated.